central/image/datastore/store/postgres/store.go

```diff
@@ -1150,6 +1153,57 @@ func (s *storeImpl) retryableGetMany(ctx context.Context, ids []string) ([]*stor
 	return elems, missingIndices, nil
 }
 
+// WalkByQuery returns the objects specified by the query
+func (s *storeImpl) WalkByQuery(ctx context.Context, q *v1.Query, fn func(image *storage.Image) error) error {
+	defer metrics.SetPostgresOperationDurationTime(time.Now(), ops.WalkByQuery, "Image")
+
+	conn, release, err := s.acquireConn(ctx, ops.WalkByQuery, "Image")
+	if err != nil {
+		return err
+	}
+	defer release()
+
+	tx, err := conn.Begin(ctx)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err := tx.Rollback(ctx); err != nil {
+			log.Errorf("error rolling back: %v", err)
+		}
+	}()
+
+	ctxWithTx := postgres.ContextWithTx(ctx, tx)
+	fetcher, closer, err := pgSearch.RunCursorQueryForSchema[storage.Image, *storage.Image](ctxWithTx, pkgSchema.ImagesSchema, q, s.db)
+	if err != nil {
+		return err
+	}
+	defer closer()
+	for {
+		rows, err := fetcher(cursorBatchSize)
+		if err != nil {
+			return pgutils.ErrNilIfNoRows(err)
+		}
+		for _, data := range rows {
+			img, exists, err := s.getFullImage(ctx, tx, data.GetId())
+			if err != nil {
+				return err
+			}
+			if !exists {
+				log.Errorf("UNEXPECTED: could not find image %q", data.GetId())
```

Comment:
a comment here that the image should be found because the get is occurring in the same transaction as the cursor might be helpful for those less fluent in sql

Reply:
You know, I think this might be an extra step in repulling the metadata, could probably just have an intermediate function where it's already expected

Reply:
yeah the first thing it does is grab the info the cursor would already have.  You just need all the other stuff like the CVEs and components.  Eliminates the need to query for the row you are actually looking at in the image table.  I assume node is simialr

Reply:
Updated to not do a double look up